File: java/spatiad-sdk/src/main/java/com/spatiad/sdk/SpatiadApiError.java
package com.spatiad.sdk;

public class SpatiadApiError extends RuntimeException {

    private final int statusCode;
    private final String code;
    private final String message;

    public SpatiadApiError(int statusCode, String code, String message) {
        super(message != null ? message : "Spatiad API error (HTTP " + statusCode + ")");
        this.statusCode = statusCode;
        this.code = code;
        this.message = message;
    }

    public int getStatusCode() {
        return statusCode;
    }

    public String getCode() {
        return code;
    }

    @Override
    public String getMessage() {
        return message;
    }
}

File: kotlin/spatiad-sdk/src/main/kotlin/com/spatiad/sdk/SpatiadApiError.kt
package com.spatiad.sdk

/**
 * Thrown when the Spatiad API returns a non-success HTTP response.
 */
class SpatiadApiError(
    val statusCode: Int,
    val code: String? = null,
    override val message: String,
) : Exception(message) {

    override fun toString(): String =
        "SpatiadApiError(statusCode=$statusCode, code=$code, message=$message)"
}
